src/en/reference-hook-tools.md

```diff
@@ -0,0 +1,564 @@
+provided as shell commands, but if you are writing your charm using Python, 
+they are wrapped by the charmhelpers module (
+[see the relevant docs for charmhelpers here](https://pythonhosted.org/charmhelpers/api/charmhelpers.core.hookenv.html))
+
+Many of the tools produce output,
+and those that do accept a `--format` flag whose value can be set to `json` or
+`yaml` as desired. If it's not specified, the format defaults to `smart`, which
+transforms the basic output as follows:
+
+
+  - strings are left untouched
+  - boolean values are converted to the strings `True` and `False`
+  - ints and floats are converted directly to strings
+  - lists of strings are converted to a single newline-separated string
+  - all other types (in general, dictionaries) are formatted as YAML
+
+Also see the [hook environment](./authors-hook-envitronment) page for further 
+details of the hook environment
+
+## Available commands:
+
+### close-port
+
+`close-port` unmarks a local system port. If the service is not exposed, it has
+no effect; otherwise the port is marked for imminent closure. It accepts the
+same flags and arguments as `open-port`.
+
+Examples:
+
+Close 1234/udp if it was open:
+
+```no-highlight
+close-port 1234/udp
+```
+
+Close port 80 if it was open:
+
+```no-highlight
+close-port 80
+```
+
+Close a range of ports:
+
+```no-highlight
+close-port 80-100
+```
+
+### config-get
+
+`config-get` returns information about the service configuration (as defined by
+the charm). If called without arguments, it returns a dictionary containing all
+config settings that are either explicitly set, or which have a non-nil default
+value. If the `--all` flag is passed, it returns a dictionary containing all
+defined config settings including nil values (for those without defaults). If
+called with a single argument, it returns the value of that config key. Missing
+config keys are reported as having a value of nil, and do not return an error.
+
+Getting the interesting bits of the config is done with:
+
+```no-highlight
+config-get
+key: some-value
+another-key: default-value
+```
+
+To get the whole config including the nulls:
+
+```no-highlight
+config-get --all
+key: some-value
+another-key: default-value
+no-default: null
+```
+
+To retrieve a specific value pass its key as argument:
+
+```no-highlight
+config-get [key]
+some-value
+```
+
+This command will also work if no value is set and no default is set or even
+if the setting doesn't exist. In both cases nothing will be returned.
+
+```no-highlight
+config-get [key-with-no-default]
+config-get [missing-key]
+```
+
+!!! Note: The above two examples are not misprints - asking for a value which
+doesn't exist or has not been set returns nothing and raises no errors.
+
+### juju-log
+
+`juju-log` writes its arguments directly to the unit's log file. All hook
+output is currently logged anyway, though this may not always be the case - If 
+it's important, please `juju-log` it.
```

Comment:
I know this is a quality of the English language but I would prefer not to create verbs like this.